lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -2042,10 +2044,12 @@ def _get_targets(self, relation) -> dict:
         """
         targets = {}
         for unit in relation.units:
-            port = relation.data[unit].get("port", 80)
-            hostname = relation.data[unit].get("hostname")
-            if hostname:
-                targets.update({unit.name: {"hostname": hostname, "port": port}})
+            unit_databag = relation.data.get(unit)
+            if unit_databag:
+                port = unit_databag.get("port", 80)
+                hostname = unit_databag.get("hostname")
+                if hostname:
+                    targets.update({unit.name: {"hostname": hostname, "port": port}})
```

Comment:
We can apply the same refactor here:

```suggestion
            if not (unit_databag := relation.data.get(unit)):
                continue

            if not (hostname := unit_databag.get("hostname")):
                continue

            port = unit_databag.get("port", 80)
            targets.update({unit.name: {"hostname": hostname, "port": port}})
```